src/javascripts/components/cookies-page.js

```diff
@@ -0,0 +1,174 @@
+import { getConsentCookie, setConsentCookie } from '../cookie-functions.js'
+import common from 'govuk-frontend/govuk/common'
+
+const nodeListForEach = common.nodeListForEach
+
+function CookiesPage ($module) {
+  this.$module = $module
+}
+
+CookiesPage.prototype.init = function () {
+  this.$cookiePage = this.$module
+
+  if (!this.$cookiePage) {
+    return
+  }
+
+  this.$cookieForms = this.$cookiePage.querySelectorAll('.js-cookies-page-form-wrapper')
+  this.$errorSummary = this.$cookiePage.querySelector('.js-cookies-page-error-summary')
+  this.$errorSummaryListItems = this.$errorSummary.querySelectorAll('.govuk-error-summary__list a')
+  this.$saveButton = this.$cookiePage.querySelector('.js-cookies-page-save')
+  this.$successNotification = this.$cookiePage.querySelector('.js-cookies-page-success')
+
+  nodeListForEach(this.$cookieForms, function ($cookieFormGroup) {
+    this.showUserPreference($cookieFormGroup, getConsentCookie())
+    $cookieFormGroup.removeAttribute('hidden')
+  }.bind(this))
+
+  this.$saveButton.addEventListener('click', this.savePreferences.bind(this))
+  this.$saveButton.removeAttribute('hidden')
+}
+
+CookiesPage.prototype.savePreferences = function () {
+  var preferences = {}
+
+  nodeListForEach(this.$cookieForms, function ($cookieForm) {
+    var cookieType = this.getCookieType($cookieForm)
+    var radios = $cookieForm.elements[cookieType]
+    var selectedItem = radios.value
+
+    if (!selectedItem) {
+      this.showErrorMessage($cookieForm)
+      return
+    }
+
+    this.hideErrorMessage($cookieForm)
+    preferences[cookieType] = radios.value === 'yes'
+  }.bind(this))
+
+  // if there are any errors, show error summary and don't submit
+  var $errors = this.$cookiePage.querySelectorAll('.govuk-error-message')
+  if ($errors.length !== 0) {
+    this.showErrorSummary($errors)
+    return
+  }
+
+  // otherwise save preferences to cookie and show success notification
+  setConsentCookie(preferences)
+  this.hideErrorSummary()
+  this.showSuccessNotification()
+}
+
+CookiesPage.prototype.showUserPreference = function ($cookieFormGroup, preferences) {
+  if (!preferences) {
+    return
+  }
+
+  var cookieType = this.getCookieType($cookieFormGroup)
+
+  if (!cookieType || preferences[cookieType] === undefined) {
+    return
+  }
+
+  var preference = preferences[cookieType]
+
+  if (preference === true) {
+    $cookieFormGroup.querySelector('input[value="yes"]').checked = true
+  } else if (preference === false) {
+    $cookieFormGroup.querySelector('input[value="no"]').checked = true
+  }
```

Comment:
If we're able to get to the cookieForm, I think this could be done by setting `RadioNodeList.value` which might be slightly neater:

```js
var radios = $cookieForm.elements[cookieType]
radios.value = preference? 'yes' : 'no'
```